frontend/javascripts/oxalis/controller/combinations/skeletontracing_plane_controller.js

```diff
@@ -340,6 +345,42 @@ function onClick(
   }
 }
 
+function onRightClick(
+  planeView: PlaneView,
+  position: Point2,
+  shiftPressed: boolean,
+  altPressed: boolean,
+  ctrlPressed: boolean,
+  plane: OrthoView,
+  isTouch: boolean,
+  event: MouseEvent,
+  showNodeContextMenuAt: (number, number, number) => void,
+) {
+  const state = Store.getState();
+  if (isMagRestrictionViolated(state)) {
+    // The current zoom value violates the specified magnification-restriction in the
+    // task type. Therefore, we abort the action here.
+    // Actually, one would need to handle more skeleton actions (e.g., deleting a node),
+    // but not all (e.g., deleting a tree from the tree tab should be allowed). Therefore,
+    // this solution is a bit of a shortcut. However, it should cover 90% of the use case
+    // for restricting the rendered magnification.
+    // See https://github.com/scalableminds/webknossos/pull/4891 for context and
+    // https://github.com/scalableminds/webknossos/issues/4838 for the follow-up issue.
+    return;
+  }
+
+  const { volume } = state.tracing;
+  if (!volume || volume.activeTool === VolumeToolEnum.MOVE) {
+    // We avoid creating nodes when in brushing mode.
+    const nodeId = maybeGetNodeIdFromPosition(planeView, position, plane, isTouch);
+    if (nodeId != null && nodeId > 0) {
+      showNodeContextMenuAt(event.pageX, event.pageY, nodeId);
+    } else {
+      setWaypoint(calculateGlobalPos(position), ctrlPressed);
+    }
```

Comment:
If I'm reading the code correctly, a context menu is opened on right click IFF the cursor is over an existing node. Is this correct? I think, it would be safer to only open the context menu when an additional modifier is pressed (e.g., shift). Otherwise, creating a new node very close to another node would not work, anymore. Also, we would lose the possibility to open context menus on spots where no node was placed (e.g., the context menu could also show "create node", "create branchpoint" there).

Reply:
> If I'm reading the code correctly, a context menu is opened on right click IFF the cursor is over an existing node. Is this correct?

Indeed this is correct :+1: 

> I think, it would be safer to only open the context menu when an additional modifier is pressed (e.g., shift). Otherwise, creating a new node very close to another node would not work, anymore. Also, we would lose the possibility to open context menus on spots where no node was placed (e.g., the context menu could also show "create node", "create branchpoint" there).

Thanks for the valuable feedback. I'll apply it now :smile_cat: